bin/header_scraper.py

```diff
@@ -33,15 +34,15 @@ def __init__(self):
 
     def addf(self, content):
         """Add function."""
-        self.functions.update(content)
+        self.functions.add(content)
 
     def addm(self, content):
         """Add macro."""
-        self.macros.update(content)
+        self.macros.add(content)
 
     def addt(self, content):
         """Add typedef."""
-        self.typedefs.update(content)
+        self.typedefs.add(content)
 
     def get_functions(self):
```

Comment:
No need for these getters. See https://stackoverflow.com/questions/2627002/whats-the-pythonic-way-to-use-getters-and-setters for discussion. Just access directly or use properties.